porch/repository/pkg/cache/repository.go

```diff
@@ -213,3 +246,60 @@ func (r *cachedRepository) pollOnce(ctx context.Context) {
 		klog.Warningf("error polling repo functions %s: %v", r.id, err)
 	}
 }
+
+func ToCachedPackageRevisionSlice(revisions []repository.PackageRevision) []*cachedPackageRevision {
+	result := make([]*cachedPackageRevision, len(revisions))
+	for i := range revisions {
+		current := &cachedPackageRevision{
+			PackageRevision:  revisions[i],
+			isLatestRevision: false,
+		}
+		result[i] = current
+	}
+	IdentifyLatestRevisions(result)
+	return result
+}
+
+func IdentifyLatestRevisions(result []*cachedPackageRevision) {
```

Comment:
Nit: Do we need to export this one also?

Reply:
ditto. will hide